Add new distro to Travis CI configuration for running leak tests

timwoj opened the pull request:
This adds a new `distro` entry to the Travis configuration that will configure the build to run the memory leak tests. The other builds still run as they were previously.

jsiwek commented:
Nice, thanks.  One additional thing that would just help save a bit of job-time would be only running the leak tests -- since we tag all the leak tests with `@TEST-GROUP: leaks`, we can get away with `btest -g leaks` to run just the leak tests.  Think that would be easy to change?  Looks like the script can just tack on `-g` to the main tests and the external tests rely on the `make all` target, but could just break that down into individual commands.

timwoj commented:
I think it would be better to add a `leaks` target to the Makefiles for zeek-testing and zeek-testing-private to avoid having to expand out 4 other targets. I'll add PRs for those.

timwoj commented:
@jsiwek I fixed it to only run the leaks tests and reworked it a little to pass the build distro along through docker instead of adding another "job" inside the script.
